x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelDefinition.java

```diff
@@ -57,35 +58,39 @@
                 p.namedObject(StrictlyParsedPreProcessor.class, n, null),
             (trainedModelDefBuilder) -> trainedModelDefBuilder.setProcessorsInOrder(true),
             PREPROCESSORS);
-        parser.declareObject(TrainedModelDefinition.Builder::setInput, (p, c) -> Input.fromXContent(p, ignoreUnknownFields), INPUT);
+        parser.declareString(TrainedModelDefinition.Builder::setModelId, TrainedModelConfig.MODEL_ID);
         return parser;
     }
 
     public static TrainedModelDefinition.Builder fromXContent(XContentParser parser, boolean lenient) throws IOException {
         return lenient ? LENIENT_PARSER.parse(parser, null) : STRICT_PARSER.parse(parser, null);
     }
 
+    public static String docId(String modelId) {
+        return NAME + "-" + modelId;
+    }
+
     private final TrainedModel trainedModel;
     private final List<PreProcessor> preProcessors;
-    private final Input input;
+    private final String modelId;
 
-    TrainedModelDefinition(TrainedModel trainedModel, List<PreProcessor> preProcessors, Input input) {
+    TrainedModelDefinition(TrainedModel trainedModel, List<PreProcessor> preProcessors, @Nullable String modelId) {
         this.trainedModel = ExceptionsHelper.requireNonNull(trainedModel, TRAINED_MODEL);
         this.preProcessors = preProcessors == null ? Collections.emptyList() : Collections.unmodifiableList(preProcessors);
-        this.input = ExceptionsHelper.requireNonNull(input, INPUT);
+        this.modelId = modelId;
```

Comment:
I added the `modelId` here so that it is trivially queriable on config deletion. That way we can just do a DBQ on the `model_id` field for the deleted model.